### PR TITLE
Feature: Transaction detail

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description
- Shows modal with transaction detail when transfer is clicked
- Modifies modal max width

Task: [DAO-820](https://aragonassociation.atlassian.net/jira/software/c/projects/DAO/boards/2?modal=detail&selectedIssue=DAO-820&sprint=52)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/utils/types.ts [hunk 1]
```diff
@@ -74,17 +59,52 @@ export type VaultToken = TokenWithMarketData & {
...
-  transferType: 'VaultDeposit' | 'VaultWithdraw';
   transferTimestamp?: string | number;
   usdValue: string;
   isPending?: boolean;
-};
+  tokenImgUrl: string;
+  tokenName: string;
+  reference?: string;
+  transaction: string;
+} & (
```
Comment (sepehr2github) [inline packages/web-app/src/utils/types.ts:104]: Looks Cool ;)

File: packages/web-app/src/utils/types.ts [hunk 2]
```diff
@@ -74,17 +59,52 @@ export type VaultToken = TokenWithMarketData & {
...
+  | {transferType: 'VaultDeposit'; sender: Address}
+  | {transferType: 'VaultWithdraw'; to: Address; proposalId: string}
```
Comment (sepehr2github) [inline packages/web-app/src/utils/types.ts]: Looks Cool ;).
One small thing, we have a transfer type enum on utils constants, Is there any way to use that type here and merge all transfers type into the one?

File: packages/web-app/src/locales/en/common.json [hunk 3]
```diff
@@ -433,5 +433,9 @@
     "cancel": "Cancel",
     "remainingTime": "Remaining time",
     "voteNow": "Vote now"
+  },
+  "transactionDetail": {
+    "title": "Transaction Detail",
+    "viewTransaction": "View on Block Explorer"
   }
 }
```
Comment (sepehr2github) [inline packages/web-app/src/locales/en/common.json:461]: Usually, all the files have an empty line by end of them, but It's not a big deal

File: packages/web-app/src/hooks/usePollTransfersPrices.tsx [hunk 4]
```diff
@@ -36,14 +36,28 @@ export const usePollTransfersPrices = (transfers: DaoTransfer[]) => {
...
+            tokenImgUrl: metadata[index]?.imgUrl || '',
             tokenSymbol: transfer.token.symbol,
             transferDate: `${formatDate(transfer.createdAt, 'relative')}`,
             transferTimestamp: transfer.createdAt,
-            transferType: transfer.__typename,
             usdValue: `$${calculatedPrice.toFixed(2)}`,
             isPending: false,
+            transaction: transfer.transaction,
+            reference: transfer.reference,
+            ...(transfer.__typename === 'VaultDeposit'
```
Comment (sepehr2github) [inline packages/web-app/src/hooks/usePollTransfersPrices.tsx]: If we can merge that enum, We can use it in several places, for example here

**Review by sepehr2github — APPROVED** [review]
Thanks, Fabrice. PR looks good to me but I left a few recommendations! Those are nice to have mostly

File: packages/web-app/src/containers/transactionDetail/index.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            transfer.transferType === 'VaultDeposit' ? '+' : '-'
+          } ${transfer.tokenAmount}`}
+          tokenSymbol={transfer.tokenSymbol}
+          tokenImageUrl={transfer.tokenImgUrl}
+          treasuryShare={transfer.usdValue}
+        />
+        {transfer.reference && (
+          <CardText
+            type="label"
+            title="Reference"
```
Comment (josemarinas) [inline packages/web-app/src/containers/transactionDetail/index.tsx]: I believe this should be a translation function `t('transactionDetail.reference')`

File: packages/web-app/src/pages/transfers.tsx [hunk 6]
```diff
@@ -58,61 +74,77 @@ const Transfers: React.FC = () => {
...
+              }
+              placeholder={t('placeHolders.searchTransfers')}
+            />
+            <div className="flex">
+              <ButtonGroup
+                bgWhite
+                defaultValue="all"
+                onChange={handleButtonGroupChange}
+              >
+                <Option value="all" label="All" />
```
Comment (josemarinas) [inline packages/web-app/src/pages/transfers.tsx]: Update labels with translations

**Review by josemarinas — CHANGES_REQUESTED** [review]
I couldn't find anything besides the lack of some translations, good job :+1:

File: packages/web-app/src/utils/types.ts (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/utils/types.ts]: Sure thing!

File: packages/web-app/src/locales/en/common.json (hunk 3 above)
Comment (Fabricevladimir) [reply packages/web-app/src/locales/en/common.json:461]: It automatically removes it for JSON files

File: packages/web-app/src/containers/transactionDetail/index.tsx (hunk 5 above)
Comment (Fabricevladimir) [reply packages/web-app/src/containers/transactionDetail/index.tsx]: Good catch thanks!

**Review by RakeshUP — APPROVED** [review]

**Review by josemarinas — APPROVED** [review]

**Pull request merged**